src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
This is outside the scope of PHPC-887. The URI's `readpreference` option is a string (e.g. `"primary"`, `"primaryPreferred"`) and only one of [several read preference URI options](https://docs.mongodb.com/manual/reference/connection-string/#read-preference-options). `MongoDB\Driver\ReadPreference` represents a complete read preference (mode, tag sets, max staleness). This is analogous to a `MongoDB\Driver\WriteConcern` representing [multiple write concern URI options](https://docs.mongodb.com/manual/reference/connection-string/#write-concern-options).